File: scala_gem_mig_testing/pipelines/abc/code/src/main/scala/io/prophecy/pipelines/abc/graph/distribute_rows.scala
package io.prophecy.pipelines.abc.graph

import io.prophecy.libs._
import io.prophecy.pipelines.abc.udfs.PipelineInitCode._
import io.prophecy.pipelines.abc.udfs.UDFs._
import io.prophecy.pipelines.abc.config.Context
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object distribute_rows {

  def apply(context: Context, in: DataFrame): (DataFrame, DataFrame) =
    (in.filter(lit(false)), in.filter(lit(true)))

}

File: custom_gem_demos_new/pipelines/gem_testing/code/gem_testing/graph/distribute_rows.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.utils import *
from prophecy.libs import typed_lit
from gem_testing.config.ConfigStore import *
from gem_testing.functions import *

def distribute_rows(spark: SparkSession, in0: DataFrame) -> (DataFrame, DataFrame):
    df1 = in0.filter((col("customer_id") <= lit(10)))
    df2 = in0.filter(((col("customer_id") > lit(10)) & (col("customer_id") < lit(90))))

    return df1, df2
